### PR TITLE
create users list page

### LINKED ISSUE #27: create users list page
Create **UsersListPage** that will match these designs: [Figma - users list page](https://www.figma.com/file/IVY7iX8XnrzpiUNH8aJOz6/dorm-social-rooms-manager?node-id=5805%3A69919&t=hq3akBYAvCUbiXa6-0)

UsersListPage should be responsive and handle 3 breakpoints:

**360px** >= screen width < **768px** → mobile
**768px** >= screen width < **1024px** → tablet
**1200px** >= screen width → desktop

### Requirements

- Use components ***[mui table component](https://mui.com/material-ui/react-table/)*** to display users list
- For page haderer and footer use components created in #12 and #6 
- **UsersListPage** should be available under `/dashboard/users-list` url
- Add different action that depends on selected users:
  - For all users, expect users with **ADMIN** privileges there should be an option to remove a user 
  - For users which role is **PENDING**

### PR DESCRIPTION
**Pull request by 235923**
closes #27

### CODE REVIEW

**Review by kaczor6418 — CHANGES_REQUESTED** [review]

**Review by kaczor6418 — CHANGES_REQUESTED** [review]

File: src/components/UserList/UserList.tsx
Comment (kaczor6418) [inline src/components/UserList/UserList.tsx:112]: now you will trigger data fetch every time you will make any changes to this component this is something bad, but as it is just a mock data that we are going to replace soon this can stay

**Review by kaczor6418 — CHANGES_REQUESTED** [review]
Grid should be adjusted below app-bar like in designs, now it is centered
image

File: src/components/UserList/UserList.types.ts
Comment (235923) [reply src/components/UserList/UserList.types.ts:5]: Props of components like TableSortLabel require the values to start with lower case

File: src/components/UserList/UserList.types.ts
Comment (kaczor6418) [inline src/components/UserList/UserList.types.ts:14]: We can keep it of now but once we merge #34 you should use generated types

File: src/components/UserList/UserList.tsx
Comment (kaczor6418) [inline src/components/UserList/UserList.tsx:41]: lets put it in one array `rowsPerPage = [5, 10, 25]`

File: src/components/UserList/UserList.tsx
Comment (kaczor6418) [inline src/components/UserList/UserList.tsx:97]: This class should be in another directory. Check the docs:
https://github.com/dorm-social-room-manager/dsrm-frontend/wiki/Project-structure#srcerrors

File: src/components/UserList/UserList.types.ts
Comment (kaczor6418) [inline src/components/UserList/UserList.types.ts:6]: lets call this `SortingDirection`. `Order` is too generic

File: src/components/UserList/UserListHead.tsx
Comment (kaczor6418) [inline src/components/UserList/UserListHead.tsx:11]: this should not be there

File: src/components/UserList/UserListHead.tsx
Comment (kaczor6418) [inline src/components/UserList/UserListHead.tsx:81]: use enum insisted of hardcoded string

File: src/components/UserList/UserListHead.tsx
Comment (kaczor6418) [inline src/components/UserList/UserListHead.tsx:72]: use enum insisted of hardcoded string

File: src/components/UserList/UserListToolbar.tsx
Comment (kaczor6418) [inline src/components/UserList/UserListToolbar.tsx:39]: In the future, this should be replaced by enum autogenerated from backend

File: src/components/UserList/UserList.tsx
Comment (kaczor6418) [inline src/components/UserList/UserList.tsx:38]: does it trigger eslint magic number rule?

**Review by kaczor6418 — CHANGES_REQUESTED** [review]
Left and right padding is too big, we should make it much smaller

**BUG:**
- If I select users with different privileges, all buttons are available. We should rather hide buttons or allow clicking it and display snackbar that you cannot remove admin users/you cannot approve non-pending users
image

File: src/components/UserList/UserList.tsx
Comment (235923) [reply src/components/UserList/UserList.tsx:41]: this triggers eslint magic number rule

File: src/components/UserList/UserList.tsx
Comment (235923) [reply src/components/UserList/UserList.tsx:38]: yes

File: src/components/UserList/UserList.tsx
Comment (kaczor6418) [reply src/components/UserList/UserList.tsx:42]: We will get rid of this rule as it triggers too many magic numbers in this project. Please add magic number to disabled rules

File: src/components/UserList/UserList.tsx
Comment (kaczor6418) [reply src/components/UserList/UserList.tsx:39]: We will get rid of this rule as it triggers too many magic numbers in this project. Please add magic number to disabled rules

**Review by kaczor6418 — CHANGES_REQUESTED** [review]

File: src/errors/FetchError.ts
Comment (235923) [reply src/errors/FetchError.ts:4]: doing it this way returns "Cannot assign to 'name' because it is a read-only property."

**Review by kaczor6418 — APPROVED** [review]
LGTM 😎

**Pull request merged**
